### PR TITLE
[iOS] Migrate tests from xamarin-macios to the repo.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Bring all tests but the ProcessManager tests to the repo. Only the
following cosmetic tests have been done:

* Follow the code style of dotnet.
* Add the OSS header.

### CODE REVIEW

**Comment by mandel-macaque** [comment]
Added as a draft PR to simplify the code review.

**Comment by mandel-macaque** [comment]
@MattGal did I add the tests in the expected location?

**Comment by premun** [comment]
@mandel-macaque test called `FormatArgumentsTest` will fail on Windows and will break the pipeline. I am not aware of any out-of-the-box solution from arcade that would enable you to skip based on platform but maybe if it's one test only, we can also mark it as skipped?

**Comment by mandel-macaque** [comment]
@premun yes, make sense, is not nice, but a valid point.

**Comment by premun** [comment]
@MattGal just FYI the tests are using NUnit: https://github.com/dotnet/xharness/issues/5

Maybe that is why there's no test results found? I see them run and it does find some XML.

I have started the migration of these to XUnit and am almost finished, so I can open PR to this branch then..

**Comment by premun** [comment]
I am finished. Maybe we want to merge the migration to XUnit first to this branch?

https://github.com/dotnet/xharness/pull/48

**Comment by MattGal** [comment]
> @MattGal just FYI the tests are using NUnit: #5
> 
> Maybe that is why there's no test results found? I see them run and it does find some XML...

Yes, I was aware of this;  I just made enough changes to get the tests to try to report to AzDO, they should actually automatically get run once they're xunit.

File: tests/Microsoft.DotNet.XHarness.iOS.Shared.Tests/CrashSnapshotReporterTests.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+using Moq;
+using NUnit.Framework;
+using Microsoft.DotNet.XHarness.iOS.Shared.Execution;
+using Microsoft.DotNet.XHarness.iOS.Shared.Execution.Mlaunch;
+using Microsoft.DotNet.XHarness.iOS.Shared.Logging;
+using Microsoft.DotNet.XHarness.iOS.Shared.Utilities;
+
+namespace Microsoft.DotNet.XHarness.iOS.Shared.Tests
+{
+    [TestFixture]
```
Comment (MattGal) [inline tests/Microsoft.DotNet.XHarness.iOS.Shared.Tests/CrashSnapshotReporterTests.cs:19]: am confused, isn't this supposed to be converted to Xunit now?

Comment (premun) [reply tests/Microsoft.DotNet.XHarness.iOS.Shared.Tests/CrashSnapshotReporterTests.cs:19]: When we close #48 it will

**Pull request merged**